tests/pipeline/test_ops.py

```diff
@@ -0,0 +1,130 @@
+from __future__ import annotations
+
+from amltk.pipeline import Choice, Node
+from amltk.pipeline.components import Sequential
+from amltk.pipeline.ops import factorize
+
+
+def test_factorize_base_case():
+    node = Node(name="hi")
+    assert list(factorize(node)) == [node]
+
+
+def test_factorize_with_no_choices_returns_same_pipeline():
+    node = Node(Node(name="n1"), Node(name="n2"), name="n0")
+    assert list(factorize(node)) == [node]
+
+
+def test_factorize_with_single_choice_same_pipeline():
+    node = Choice(Node(name="n1"), name="c1")
+    assert list(factorize(node)) == [node]
+
+
+def test_factorize_with_two_choices_returns_both_pipelines():
+    n1 = Node(name="n1")
+    n2 = Node(name="n2")
+    choice = Choice(n1, n2, name="c1")
+    expected = [
+        Choice(n1, name="c1"),
+        Choice(n2, name="c1"),
+    ]
+    assert list(factorize(choice)) == expected
+
+
+def test_nested_choice_returns_possible_pipelines():
```

Comment:
Does this mean it factorizes all choices? I somehow assumed that this only factorizes the first choice (or the current choice). If so, could we control this, like "factories up to depth 5".  Also connected to next test.

Reply:
Sure why not, can you give a specific use case I could include in the documentation for that?

Reply:
Something like: You have a pipeline including all your algorithms with a choice at level 1 splitting between these algorithms. Now, you could use this function to get the pipeline per algorithm, allowing you to re-use the general pipeline easily for more specific tasks.